spikeinterface/sortingcomponents/peak_localization.py

```diff
@@ -3,54 +3,85 @@
 from spikeinterface.core.job_tools import ChunkRecordingExecutor, _shared_job_kwargs_doc
 from spikeinterface.toolkit import get_noise_levels, get_channel_distances
 
+import scipy.optimize
+
+from ..toolkit import get_chunk_with_margin
+
+from ..toolkit.postprocessing.unit_localization import (dtype_localize_by_method,
+    possible_localization_methods, 
+    estimate_distance_error
+    )
+
+
+def init_kwargs_dict(method, method_kwargs):
+    if method == 'center_of_mass':
+        method_kwargs_ = dict(local_radius_um=150)
+    elif method == 'monopolar_triangulation':
+        method_kwargs_ = dict(local_radius_um=150, max_distance_um=1000)
+    method_kwargs_.update(method_kwargs)
+    return method_kwargs_
+    
+
+def localize_peaks(recording, peaks, ms_before=0.1, ms_after=0.3,
+    method='center_of_mass', method_kwargs={},
+    **job_kwargs):
+                   #~ local_radius_um=150, 
 
-def localize_peaks(recording, peaks, method='center_of_mass',
-                   local_radius_um=150, ms_before=0.3, ms_after=0.6,
-                   **job_kwargs):
     """
-    Localize peak (spike) in 2D or 3D depending the probe.ndim of the recording.
+    Localize peak (spike) in 2D or 3D depending the method.
+    When a probe is 2D then:
+       * X is axis 0 of the probe
+       * Y is axis 1 of the probe
+       * Z is orthogonal to the plane of the probe
 
     Parameters
     ----------
     recording: RecordingExtractor
         The recording extractor object
     peaks: numpy
         peak vector given by detect_peaks() in "compact_numpy" way.
-    method: str
-        Method to be used ('center_of_mass')
-    local_radius_um: float
-        Radius in micrometer to make neihgborhood for channel
-        around the peak
     ms_before: float
         The left window before a peak in millisecond
     ms_after: float
         The left window before a peak in millisecond
-    {}
+    method: str
+        Method to be used ('center_of_mass')
+    method_kwargs: dict of kwargs method
+        'center_of_mass':
+            * local_radius_um: float
+                For channel sparsity
+        'monopolar_triangulation' also have:
+            * local_radius_um: float
+                For channel sparsity
+            * max_distance_um: float default 1000
```

Comment:
this could be estimated from the locations no?

Reply:
No. It is the maximum of a unit from the best channel. It is to make boundary on x, y, z nd alpha